pinot-common/src/main/java/com/linkedin/pinot/common/metadata/segment/LLCRealtimeSegmentZKMetadata.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.common.metadata.segment;
+
+import java.util.Map;
+import org.apache.helix.ZNRecord;
+import static com.linkedin.pinot.common.utils.EqualityUtils.hashCodeOf;
+import static com.linkedin.pinot.common.utils.EqualityUtils.isEqual;
+import static com.linkedin.pinot.common.utils.EqualityUtils.isNullOrNotSameClass;
+import static com.linkedin.pinot.common.utils.EqualityUtils.isSameReference;
+
+
+public class LLCRealtimeSegmentZKMetadata extends RealtimeSegmentZKMetadata {
+  private static final String START_OFFSET = "segment.realtime.startOffset";
+  private static final String END_OFFSET = "segment.realtime.endOffset";
+  private static final String NUM_REPLICAS = "segment.realtime.numReplicas";
+
+  private long _startOffset;
+  private long _endOffset;
+  private int _numReplicas;
+
+  public LLCRealtimeSegmentZKMetadata() {
+    super();
+  }
+
+  public LLCRealtimeSegmentZKMetadata(ZNRecord znRecord) {
+    super(znRecord);
+    _startOffset = Long.valueOf(znRecord.getSimpleField(START_OFFSET));
+    _numReplicas = Byte.valueOf(znRecord.getSimpleField(NUM_REPLICAS));
```

Comment:
Should be int to match with the type. In the unlikely case of 128 replicas this will fail.